blogs/2020/07/27/containers-edu.md

```diff
@@ -0,0 +1,223 @@
+
+#### Dockerfile
+
+Docker can build images automatically by reading the instructions from your [Dockerfile](https://docs.docker.com/engine/reference/builder/). This file will at the very least have a base image specified using “FROM,” and it can include command-line instructions as well.
+
+Here’s what the Dockerfile looks like in our Python app:
+
+![Python dockerfile in app](19-dockerfile.png)
+
+We technically didn’t need to use the **Remote – Containers: Add Development Container Configuration Files…** command- we could have created a .devcontainer folder and added a `devcontainer.json` and Dockerfile to it. The great part of the Remote – Containers extension is that it has configuration files setup for us so that we can simply call a command and save even more time.
+
+### Opening the container
+
+Once our container configuration files were added, we got a notification that our folder now has a `devcontainer.json` file. Let’s select **Reopen in Container**:
+
+![Reopen in container notification](21-reopen-notif.png)
+
+A new instance of VS Code launches, and we can see our image is being built and our app is starting in our dev container:
+
+![Relaunch VS Code to start dev container](22-reload-starting.png)
+
+Once our container is built, we can see we have all the same files, but now the bottom left indicator reads, "Dev Container: Python 3,” signifying we’re inside our container. We can run our code with ease, regardless of which version of Python (if any) we have installed on our local machine!
+
+![Run Python code in dev container](24-run-code.png)
+
+### Creating and sharing assignments
+
+Now that you have a dev container, you can upload this to a repository in which your students can access it for their assignments. You could include assignment instructions in a README, any base or template code needed for the assignment, and then a .devcontainer so that they all have a consistent development experience. This can all be uploaded as a single repo.
+
+Make sure you have [Git installed](https://git-scm.com/downloads). You can [initialize a new Git repository](https://code.visualstudio.com/docs/editor/versioncontrol#_initialize-a-repository) from your container project in VS Code. Alternatively, you can use `git init` from your project’s directory in the command line.
+
+Following the guide [here](https://kbroman.org/github_tutorial/pages/init.html), the next step is to connect your local repository to a [new repo](https://github.com/new) from the GitHub desktop client:
+
+![Create a new GitHub repo online](25-create-repo.png)
+
+GitHub gives us some tips about how to add code to our newly created repo.
+
+Let’s connect our local Git repo (the container project on our computer) to the remote repo (the GitHub repo we created in the browser):
+
+![Connect Git repos in command prompt](26-git-setup.png)
```

Comment:
Suggestion: cut empty space at bottom of screenshot

Reply:
Makes sense- just cropped.